### DISCUSSION TITLE
Ability for Leader Key style keys

### DISCUSSION THREAD

**Discussion by mikevalstar** [JohannesKlauss/react-hotkeys-hook#1138] [Ideas]
similar to how VI does it, but also applications like Atlassian Confluence, you can do a leader key style navigation (also called gmail style sequences in mousetrap): 
![image](https://github.com/JohannesKlauss/react-hotkeys-hook/assets/1720/69f8f5e8-fdd9-4c00-8cb1-426d746d5289)

**Comment by JohannesKlauss** [comment]
Yes, I like that idea very much. I need to get version 5 through the door first, though, to have a stable key parsing and getting rid of the whole keyboard layout confusion.

**Comment by JohannesKlauss** [comment]
Do you know of a OSS implementation of this listening style? Does it have a threshold on when the first key gets abandoned if no other key gets pressed?

**Reply by mikevalstar** [reply]
https://github.com/ccampbell/mousetrap

It looks like it just keeps a running sequence until it hits a match: 

https://github.com/ccampbell/mousetrap/blob/master/mousetrap.js#L681
